pkg/rbac/cache.go

```diff
@@ -13,11 +15,16 @@ type Cache struct {
 	tokenReviews        map[string]*tokenReviewResult
 	tokenReviewsPending map[string][]chan *tokenReviewResult
 	tokenReviewsLock    sync.Mutex
+	shared              sharedList
+	sharedLock          sync.Mutex
+	pool                pgxpoolmock.PgxPool
```

Comment:
This should be a pointer to the pool to ensure that we only have a single connection pool. We'll need to update `db.GetConnection()`
```suggestion
	pool                *pgxpoolmock.PgxPool
```

Reply:
Issue with converting pgxpool -> pgxpoolmock in connection.go was `ConnectConfig() ` isn't part of pgxpoolmock package.
![image](https://user-images.githubusercontent.com/19152767/173106343-5d1b4297-b475-4a99-9e3b-83fbac556095.png)